test/jdk/java/awt/Robot/RobotModifierMaskTest.java

```diff
@@ -0,0 +1,231 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import sun.awt.OSInfo;
+
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.StringSelection;
+import java.awt.event.KeyEvent;
+
+/*
+ * @test
+ * @bug 8302618
+ * @key headful
+ * @modules java.desktop/sun.awt
+ * @requires (os.family == "mac")
+ * @summary To test if modifier keys work properly,
+ *          when manual mouse move and Robot's Key Event occur simultaneously.
+ */
+public class RobotModifierMaskTest {
+
+    private static Robot robot;
+    private static JFrame jFrame;
+    private static JTextArea jTextArea;
+
+    private static StringBuffer errorLog = new StringBuffer();
+    private static final String EXPECTED_RESULT_SHIFT = "AAAAA";
+    private static final String EXPECTED_RESULT_CAPS = "AaAaAa";
+    private static final String EXPECTED_RESULT_META = "AAA";
+    private static final String EXPECTED_RESULT_ALT = "\u00e5\u00e5\u00e5";
+    private static final int EXPECTED_CARET_POS_CTRL = 0;
+
+    private static final String INSTRUCTIONS = """
+            This test is a semi-automatic test which checks the effect of typing modifier keys
+            through a Robot.\n
+            It tests the following key modifiers - Shift, Caps, Control, Option and Command.
+            It needs to be checked for following two scenarios \n
+
+            CASE 1 : Run the test as an automated test and let the Robot go through all the test cases.\n
+            CASE 2 : Run the test in semi-automated mode. While the Robot in typing,
+                     manually move the mouse (without clicking/dragging). Check if the test Passes or Fails.\n\n
+
+            NOTE: User doesn't need to compare the actual vs expected result in Case 2.
+            The test compares it.""";
+
+    public static void main(String[] args) throws Exception {
+        if (OSInfo.getOSType() != OSInfo.OSType.MACOSX) {
+            System.out.println("This test is for MacOS platform only");
+            return;
+        }
+
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+            robot.setAutoDelay(200);
+
+            SwingUtilities.invokeAndWait(() -> createTestUI());
```

Comment:
I don't insist, yet IDE usually show a warning that such a lambda expression can be replaced with a method reference.